File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        maxArea = -float("inf")
        while left < right:
            y = min(height[left], height[right])
            x = right - left
            maxArea = max(maxArea, x * y)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return maxArea

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int maxArea = 0;
        while(left < right){
            int area = (right - left) * min(height[left], height[right]);
            maxArea = max(area, maxArea);
            if(height[left] > height[right]) 
                right--;
            else   
                left++;
        }
        return maxArea;
    }
};